pkgs/tools/package-management/apt/default.nix

```diff
@@ -16,17 +17,17 @@
 stdenv.mkDerivation rec {
   pname = "apt";
 
-  version = "1.4.6";
+  version = "1.8.4";
 
-  src = fetchzip {
-    url = "https://launchpad.net/ubuntu/+archive/primary/+files/apt_${version}.tar.xz";
-    sha256 = "0ahwhmscrmnpvl1r732wg93dzkhv8c1sph2yrqgsrhr73c1616ix";
+  src = fetchurl {
+    url = "http://deb.debian.org/debian/pool/main/a/apt/apt_${version}.tar.xz";
+    sha256 = "0gn4srqaaym85gc8nldqkv01477kdwr136an2nlpbdrsbx3y83zl";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkgconfig cmake ninja ];
 
   buildInputs = [
-    cmake perlPackages.perl curl gtest lzma bzip2 lz4 db dpkg libxslt.bin
+    perlPackages.perl curl gtest gnutls lzma bzip2 lz4 zstd libseccomp udev db dpkg libxslt.bin
```

Comment:
`perlPackages.perl` (and Po4a) + `gtest` + `libxslt` are only needed at build time and should be in `nativeBuildInputs`